client/src/lib/hooks/useConnection.ts

```diff
@@ -297,7 +297,16 @@ export function useConnection({
         bearerToken || (await serverAuthProvider.tokens())?.access_token;
       if (token) {
         const authHeaderName = headerName || "Authorization";
-        headers[authHeaderName] = `Bearer ${token}`;
+        
+        // Add custom header name as a special request header to let the server know which header to pass through 
+        if (authHeaderName.toLowerCase() !== "authorization") {
+          headers[authHeaderName] = token;
+          if (headerName) {
+            headers["x-custom-auth-header"] = headerName;
+          }
```

Comment:
In this case, `authHeaderName` is already known to be something other than `undefined`, `null`, or "Authorization". There's no need to null-check or to refer to `headerName` rather than `authHeaderName`. 

```suggestion
          headers["x-custom-auth-header"] = authHeaderName;
```

Reply:
Removed.